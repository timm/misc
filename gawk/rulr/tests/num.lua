local r=require"rulr"
local l,NUM= r.lib,r.NUM

math.randomseed(1234567891)
num=NUM.new()
for _ =1,1000 do num:add(math.random()^0.5) end

for x=0,1,0.02 do 
  local a=l.rnd(num:area(x),2)
  print(x, a, l.rnd(num:range(x)), ("*"):rep(a * 50 // 1)) end

assert(0.671 == l.rnd(num.mu,3),"bad mu")
assert(0.237 == l.rnd(num.sd,3),"bad sd")