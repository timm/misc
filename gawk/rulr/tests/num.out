% lua num.lua

0.0	0.0	1	
0.02	0.0	1	
0.04	0.0	1	
0.06	0.01	1	
0.08	0.01	1	
0.1	0.01	1	
0.12	0.01	1	
0.14	0.01	1	
0.16	0.02	1	*
0.18	0.02	1	*
0.2	0.02	1	*
0.22	0.03	1	*
0.24	0.03	1	*
0.26	0.04	1	**
0.28	0.05	1	**
0.3	0.06	1	***
0.32	0.07	1	***
0.34	0.08	1	****
0.36	0.1	1	*****
0.38	0.11	1	*****
0.4	0.13	1	******
0.42	0.15	2	*******
0.44	0.17	2	********
0.46	0.19	2	*********
0.48	0.22	2	***********
0.5	0.24	2	************
0.52	0.27	2	*************
0.54	0.3	3	***************
0.56	0.33	3	****************
0.58	0.36	3	******************
0.6	0.39	3	*******************
0.62	0.42	3	*********************
0.64	0.45	4	**********************
0.66	0.48	4	************************
0.68	0.51	4	*************************
0.7	0.55	4	***************************
0.72	0.58	5	****************************
0.74	0.61	5	******************************
0.76	0.64	5	********************************
0.78	0.67	5	*********************************
0.8	0.7	5	***********************************
0.82	0.73	6	************************************
0.84	0.76	6	**************************************
0.86	0.78	6	***************************************
0.88	0.81	6	****************************************
0.9	0.83	6	*****************************************
0.92	0.85	6	******************************************
0.94	0.87	7	*******************************************
0.96	0.89	7	********************************************
0.98	0.9	7	*********************************************
